### PR TITLE
Migrate from npm to pnpm

### PR DESCRIPTION
**Pull request by CoyoteClear340**
This PR migrates the project from npm to pnpm. It replaces the npm-lock.json file with pnpm-lock.yaml, updates references in the README, and ensures the project uses pnpm for better performance and consistency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,8 +108,10 @@ To get a local copy up and running follow these simple example steps.
 
 * install dependencies and set up frontend
   ```sh
-  npm install
-  node start
+  # Install dependencies using  pnpm
+  pnpm instal
```
Comment (Majestic9169) [inline README.md]: `pnpm install`

**Review by Majestic9169 — CHANGES_REQUESTED** [review]
Feel free to delete the `package-lock.json` now, having both that and `pnpm-lock` is confusing and unnecessary 

Also update your `pnpm-lock` and `package.json` with `pnpm update` 

And fix that typo in the `README.md`

**Comment by Majestic9169** [comment]
Also next time you create a pr you can reference the issue by adding `#<issue_number>` in the pr description. This makes it easier for me to assign the issue to you and keep track of things.

**Comment by CoyoteClear340** [comment]
Oh, okay! I'll make sure to include the issue number in the PR description from now on. Thank you for pointing it out!

**Review by Majestic9169 — APPROVED** [review]
looks good

**Pull request merged**